src/main/java/core/basesyntax/service/AbstractService.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Warehouse;
+import core.basesyntax.model.entities.Product;
+
+public abstract class AbstractService<T extends Product> {
```

Comment:
let's create an interface instead of an abstract class with the methods that we will implement

Reply:
Every service needs a warehouse to work with, so I cant remove an abstract class. Or I need to create an interface as well?

Reply:
we can address the warehouse directly because it is static - we don't need the warehouse field anywhere